client/protocol.go

```diff
@@ -379,38 +399,25 @@ func (c *protocolClient) newSeqNum() uint64 {
 // Send fetch cluster requests to all servers
 // Note: The fetched cluster may still be outdated if `linearizable` is false
 func (c *protocolClient) fetchCluster(linearizable bool) (*curpapi.FetchClusterResponse, error) {
-	resCh := make(chan *curpapi.FetchClusterResponse)
-	errCh := make(chan error)
-
 	timeout := c.getBackoff()
 	retryCnt := c.config.RetryCount
-	for i := 0; i < retryCnt; i++ {
-		retryTimeout := timeout.nextRetry()
-		for _, conn := range c.connects {
-			conn := conn
-			go func() {
-				protocolClient := curpapi.NewProtocolClient(conn)
-				ctx, cancel := context.WithTimeout(context.Background(), retryTimeout)
-				defer cancel()
-				res, err := protocolClient.FetchCluster(ctx, &curpapi.FetchClusterRequest{Linearizable: linearizable})
-				if err != nil {
-					errCh <- err
-					return
-				}
-				resCh <- res
-			}()
-		}
-	}
 
 	var maxTerm uint64 = 0
 	var res *curpapi.FetchClusterResponse
 	okCnt := 0
 	majorityCnt := len(c.connects)/2 + 1
 
-OuterLoop:
-	for i := 0; i < len(c.connects); i++ {
-		select {
-		case r := <-resCh:
+	for i := 0; i < retryCnt; i++ {
+		retryTimeout := timeout.nextRetry()
+		for _, conn := range c.connects {
+			protocolClient := curpapi.NewProtocolClient(conn)
+			ctx, cancel := context.WithTimeout(context.Background(), retryTimeout)
+			defer cancel()
+			r, err := protocolClient.FetchCluster(ctx, &curpapi.FetchClusterRequest{Linearizable: linearizable})
+			if err != nil {
+				c.logger.Warn("fetch cluster error", zap.Error(err))
+				continue
+			}
```

Comment:
This refactoring will cause `FetchCluster` to fail to be concurrent, slowing down the request speed

Reply:
Thanks for the suggestion, I've changed it to be concurrent.